server/src/test/java/org/elasticsearch/cluster/health/ClusterIndexHealthTests.java

```diff
@@ -141,187 +140,54 @@ protected Predicate<String> getRandomFieldsExcludeFilter() {
-                    instance.getShards()
-                );
-            case "status":
-                ClusterHealthStatus status = randomFrom(
-                    Arrays.stream(ClusterHealthStatus.values()).filter(value -> value.equals(instance.getStatus()) == false).toList()
-                );
-                return new ClusterIndexHealth(
-                    instance.getIndex(),
-                    instance.getNumberOfShards(),
-                    instance.getNumberOfReplicas(),
-                    instance.getActiveShards(),
-                    instance.getRelocatingShards(),
-                    instance.getInitializingShards(),
-                    instance.getUnassignedShards(),
-                    instance.getUnassignedPrimaryShards(),
-                    instance.getActivePrimaryShards(),
-                    status,
-                    instance.getShards()
-                );
-            case "shards":
-                Map<Integer, ClusterShardHealth> map;
+        String index = instance.getIndex();
+        int numberOfShards = instance.getNumberOfShards();
+        int numberOfReplicas = instance.getNumberOfReplicas();
+        int activeShards = instance.getActiveShards();
+        int relocatingShards = instance.getRelocatingShards();
+        int initializingShards = instance.getInitializingShards();
+        int unassignedShards = instance.getUnassignedShards();
+        int unassignedPrimaryShards = instance.getUnassignedPrimaryShards();
+        int activePrimaryShards = instance.getActivePrimaryShards();
+        ClusterHealthStatus status = instance.getStatus();
+        Map<Integer, ClusterShardHealth> shards = instance.getShards();
+
+        switch (randomIntBetween(0, 10)) {
+            case 0 -> index = instance.getIndex() + randomAlphaOfLengthBetween(2, 5);
+            case 1 -> numberOfShards = instance.getNumberOfShards() + between(1, 10);
+            case 2 -> numberOfReplicas = instance.getNumberOfReplicas() + between(1, 10);
+            case 3 -> activeShards = instance.getActiveShards() + between(1, 10);
+            case 4 -> relocatingShards = instance.getRelocatingShards() + between(1, 10);
+            case 5 -> initializingShards = instance.getInitializingShards() + between(1, 10);
+            case 6 -> unassignedShards = instance.getUnassignedShards() + between(1, 10);
+            case 7 -> unassignedPrimaryShards = instance.getUnassignedPrimaryShards() + between(1, 10);
+            case 8 -> activePrimaryShards = instance.getActivePrimaryShards() + between(1, 10);
+            case 9 -> status = randomValueOtherThan(instance.getStatus(), () -> randomFrom(ClusterHealthStatus.values()));
+            case 10 -> {
                 if (instance.getShards().isEmpty()) {
-                    map = Collections.singletonMap(0, ClusterShardHealthTests.randomShardHealth(0));
+                    shards = Collections.singletonMap(0, ClusterShardHealthTests.randomShardHealth(0));
```

Comment:
Sorry, I missed this in my first review. I usually prefer `Map.of()` instead of any of the methods on `Collections`. The resulting map is the same, but `Map.of()` makes it easier to add new extra entries in the parameters if we ever need to (and a tiny bonus is that it's a little bit shorter). Again, this is a personal preference, so I'm fine with leaving this as is for this PR. Just wanted to tell you for the future.

Reply:
Have changed that :-) Anything that makes the code more readable is good!